### DISCUSSION TITLE
Drawing on the map

### DISCUSSION THREAD

**Discussion by evertheylen** [osmandapp/OsmAnd#11577] [Ideas]
I've been trying the OsmAnd app, it looks nice and packed with useful features. As the title suggests, I've been looking for one feature in particular: the ability to (freely) draw on the map. I envision it like an extra layer on top of the normal layers, that moves together with the rest of the map. In other words, not just taking a screenshot and drawing on the screenshot. I have an android tablet with pen input, but presumably this feature would work with "normal" touch too. A few questions on this:

1. I haven't found any mention of a feature like this. Is this correct or did I miss something?
2. I am developer myself and saw there are plugins. Would this be doable as a plugin?
3. Anyone else interested in this? I'd also be willing to offer some kind of bounty.

**Comment by tommycrock** [comment]
Hi Evert

Although it isn't exactly designed to do this, you may have all the functionality you want using the "plan a route" mode. You can draw lines across the map, use new segment to start new lines and then save. If you wanted different colours you could save a different track with the lines that should be that colour and change the colour. You couldn't do filled areas I don't think, but you could use custom POI icons.

**Comment by SimonMellerin** [comment]
I'd also love to see a plugin that could let me plan a trip like:

- add favorite places (already available, I know, but the idea here is to have a complete "plan a trip" experience)
- freely draw on map (free drawing and/or polygons, circle)
- fill area with custom color
- write text on map

Could really be awesome!

Also, I've never see such a functionality on other android map app

**Reply by mariush444** [reply]
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/a2064262-9f7d-4b31-947e-abc6c016710a"  width="300" height="auto"/> </td>
  </tr>
</table>
Try oruxmap

**Comment by ezemskov** [comment]
Thanks for inspiration gentlemen.
As a user, I've been always happy with just point markers, but agree - ability to add some freeform drawings on the map would be useful (considering resolution of modern screens).
Might try to write a plugin for that, as an exercise in Android development. 
Couple things that  immediately into mind : 
  - if [freeform] drawing is raster - how would/should it scale when map zoom level changes after drawing ?
   - All point markers (and, as I understand, planned routes) are stored as .gpx. Wonder what would be an appropriate way to store these raster drawings (as image files of somehow limited dimensions ?)

**Reply by SimonMellerin** [reply]
thanks for your answer @ezemskov , happy to see this moving :heart_eyes: 

Why not some geojson to display and store these data ? 

Geojson will also permit to easily edit these shapes, while it will be complicated with raster
